marshal/marshal.go

```diff
@@ -315,62 +315,72 @@ func (m *Marshaler) ClaimPartition(topicName string, partID int) bool {
 // Heartbeat will send an update for other people to know that we're still alive and
 // still owning this partition. Returns an error if anything has gone wrong (at which
 // point we can no longer assert we have the lock).
-func (m *Marshaler) Heartbeat(topicName string, partID int, CurrentOffset int64) error {
+func (m *Marshaler) Heartbeat(topicName string, partID int, offset int64) error {
 	topic, err := m.getClaimedPartitionState(topicName, partID)
 	if err != nil {
 		return err
 	}
 
+	// Attempt to commit offset, this is best-effort and we don't care if it fails
+	// since the canonical storage is in the heartbeat
+	if err := m.CommitOffsets(topicName, partID, offset); err != nil {
+		log.Warningf("[%s:%d] failed to commit offset during heartbeat: %s",
+			topicName, partID, err)
+	}
+
 	// All good, let's heartbeat
 	cl := &msgHeartbeat{
 		msgBase:       *m.msgBase(topicName, partID),
-		CurrentOffset: CurrentOffset,
+		CurrentOffset: offset,
 	}
 	_, err = m.cluster.producer.Produce(MarshalTopic, int32(topic.claimPartition),
 		&proto.Message{Value: []byte(cl.Encode())})
 	if err != nil {
+		log.Errorf("[%s:%d] failed to send heartbeat message to Kafka: %s",
```

Comment:
The caller knows what they passed in so I didn't think it necessary, I also figured that most callers who get errors will print out the parameters. Something like "failed committing offset to topic:offset, got $error".

I could go either way though.